src/IrcCommands/Commands/Nick.cpp

```diff
@@ -0,0 +1,61 @@
+#include <iostream>
+#include <list>
+#include "../../Client/Client.hpp"
+#include "../../Server/Server.hpp"
+#include "../../includes/CONSTANTS.hpp"
+#include "../../includes/types.hpp"
+#include "../IrcCommands.hpp"
+
+/**
+ * @brief functin to create or change the nickname of a certain client
+ *
+ * TODO:
+ * (1) think about client feedback in case of success
+ * (2) think about if all other clients should get a feedback in case
+ * of the client has changed his nick (maybe only as a channel notice)
+ *
+ * @return 0, in case of an error it returns error codes:
+ * ERR_NEEDMOREPARAMS (461)
+ * ERR_ALREADYREGISTERED (462)
+ * ERR_PASSWDMISMATCH (464)
+ *  ->chars that introduce ambiguity in other commands 
+ *    (prefix, tags, channels, spaces)
+ *
+ * not in use: 
+ * ERR_NICKCOLLISION (436) ->only relevant for multi-server setup
+ */
+int IrcCommands::nick(Server& base, const struct cmd_obj& cmd,
+                      int fd_curr_client) {
+  std::list<Client>::iterator it_client = base._client_list.begin();
+  for (; it_client != base._client_list.end(); it_client ++) {
+    if (it_client->get_client_fd() == fd_curr_client)
+      break;
+  }
+
```

Comment:
when i get this right, this operation happens at the beginning of every function. I think we should move it into a dedicated utility function